cmd/cosign/cli/generate/generate_key_pair.go

```diff
@@ -55,13 +54,17 @@ func GenerateKeyPairCmd(ctx context.Context, kmsVal string, args []string) error
 		if err != nil {
 			return err
 		}
-		if err := os.WriteFile("cosign.pub", pemBytes, 0600); err != nil {
+		if err := os.WriteFile(name+".pub", pemBytes, 0600); err != nil {
 			return err
 		}
-		fmt.Fprintln(os.Stderr, "Public key written to cosign.pub")
+		fmt.Fprintln(os.Stderr, "Public key written to "+name+".pub")
 		return nil
 	}
 
+	if name == "" {
```

Comment:
why don't you just specify this value as the flag's default value, wouldn't it be better?

```go
cmd.Flags().StringVar(&o.Name, "name", "cosign", "adds a choosen name to your .pub and .key files")
```